drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c

```diff
@@ -529,43 +668,43 @@ static int ieee802154_cc13xx_cc26xx_subg_tx(const struct device *dev,
 					    struct net_buf *buf)
 {
 	struct ieee802154_cc13xx_cc26xx_subg_data *drv_data = dev->data;
+	bool was_rx_on = false;
 	RF_EventMask events;
-	int ret;
+	int ret = 0;
 
 	if (mode != IEEE802154_TX_MODE_DIRECT) {
 		/* For backwards compatibility we only log an error but do not bail. */
 		NET_ERR("TX mode %d not supported - sending directly instead.", mode);
 	}
 
-	k_mutex_lock(&drv_data->tx_mutex, K_FOREVER);
+	if (k_sem_take(&drv_data->lock, K_FOREVER)) {
+		return -EIO;
+	}
 
-	/* Prepend data with the SUN FSK PHY header,
-	 * see IEEE 802.15.4, section 19.2.4.
-	 */
+	if (!drv_data->is_up) {
+		ret = -ENETDOWN;
+		goto out;
+	}
+
+	was_rx_on = drv_data->cmd_prop_rx_adv.status == ACTIVE;
```

Comment:
This is incorrect: the `*_tx()` function is called right after `*_cca()`, which calls `drv_stop_rx()` and this statement will be false.

I think you need to record `was_rx_on`'s state from before the `*_cca()` call into a global variable and use that instead when CCA is used, otherwise `drv_start_rx()` will never get called after a successful packet transmission.

Reply:
True. I really got this all wrong because I moved those changes up from much further downstream in my dev branch. :-(

Reply:
Actually this can be fixed locally. The new driver API spec is very clear on that point, I wrote it myself but didn't implement it: Unless an RX slot is configured (which this version of the driver does not yet support - although it already exists in my dev branch), the receiver must always be on while the interface is "UP". So no need to keep that local state at all. I can simply (gracefully) switch the receiver off iif it is on and then switch it unconditionally on again after sending finishes.

Reply:
Fixed.